contracts/LimitOrderSwap.sol

```diff
@@ -70,9 +70,62 @@ contract LimitOrderSwap is ILimitOrderSwap, Ownable, TokenCollector, EIP712 {
         _fillLimitOrder(order, makerSignature, takerParams, true);
     }
 
+    /// @inheritdoc ILimitOrderSwap
+    function fillLimitOrderGroup(
+        LimitOrder[] calldata orders,
+        bytes[] calldata makerSignatures,
+        uint256[] calldata makerTokenAmounts,
+        address[] calldata profitTokens,
+        uint256 unwrapAmount
+    ) external payable override {
+        if (orders.length != makerSignatures.length || orders.length != makerTokenAmounts.length) revert InvalidParams();
+
+        // validate orders and calculate takingAmounts
+        uint256[] memory takerTokenAmounts = new uint256[](orders.length);
+        for (uint256 i = 0; i < orders.length; ++i) {
+            LimitOrder calldata order = orders[i];
+            uint256 makingAmount = makerTokenAmounts[i];
+
+            (bytes32 orderHash, uint256 orderFilledAmount) = _validateOrder(order, makerSignatures[i]);
+            {
+                uint256 orderAvailableAmount = order.makerTokenAmount - orderFilledAmount;
+                if (makingAmount > orderAvailableAmount) revert NotEnoughForFill();
+                takerTokenAmounts[i] = ((makingAmount * order.takerTokenAmount) / order.makerTokenAmount);
```

Comment:
Do we check zero amount here?
`if (takerTokenAmounts[i] == 0 && makingAmount == 0) revert ZeroTokenAmount();`

Reply:
Added in a3af4d4